src/test/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/component/LastKnownAddressComponentTest.kt

```diff
@@ -0,0 +1,117 @@
+    assertThat(lastKnownAddressId, present())
+
+    val updatedRecall = authenticatedClient.getRecall(originalRecall.recallId)
+    assertThat(originalRecall.lastKnownAddresses, isEmpty)
+
+    val lastKnownAddresses = updatedRecall.lastKnownAddresses
+    assertThat(lastKnownAddresses.size, equalTo(1))
+    assertThat(lastKnownAddresses.first().line1, equalTo("address line 1"))
+    assertThat(lastKnownAddresses.first().line2, equalTo("some line 2"))
+    assertThat(lastKnownAddresses.first().town, equalTo("some town"))
+    assertThat(lastKnownAddresses.first().postcode, equalTo("the postcode"))
+    assertThat(lastKnownAddresses.first().index, equalTo(1))
+    assertThat(lastKnownAddresses.first().source.toString(), equalTo("MANUAL"))
+    assertThat(lastKnownAddresses.first().lastKnownAddressId.toString(), equalTo(lastKnownAddressId.value.toString()))
+  }
+
+  @Test
+  fun `adding 2 LastKnownAddresses for the same recall, optional properties can be null, both will be returned on the recall with incrementing index`() {
+    val originalRecall = authenticatedClient.bookRecall(bookRecallRequest)
+    val firstLastKnownAddressRequest = CreateLastKnownAddressRequest(
+      originalRecall.recallId,
+      "address line 1",
+      "some line 2",
+      "first town",
+      "the postcode",
+      AddressSource.LOOKUP
+    )
+
+    val lastKnownAddressId1 = authenticatedClient.lastKnownAddress(firstLastKnownAddressRequest, CREATED, LastKnownAddressId::class.java)
+    assertThat(lastKnownAddressId1, present())
+
+    val secondLastKnownAddressRequest = firstLastKnownAddressRequest.copy(
+      line2 = null,
+      town = "second town",
+      postcode = null,
+      source = AddressSource.MANUAL
+    )
+    val lastKnownAddressId2 = authenticatedClient.lastKnownAddress(secondLastKnownAddressRequest, CREATED, LastKnownAddressId::class.java)
+    assertThat(lastKnownAddressId2, present())
+    assertThat(lastKnownAddressId1, !equalTo(lastKnownAddressId2))
+
+    val updatedRecall = authenticatedClient.getRecall(originalRecall.recallId)
+    assertThat(originalRecall.lastKnownAddresses, isEmpty)
+
+    val lastKnownAddresses = updatedRecall.lastKnownAddresses
+    assertThat(lastKnownAddresses.size, equalTo(2))
+
+    assertThat(lastKnownAddresses[0].index, !equalTo(lastKnownAddresses[1].index))
+    // There is no contract for the order of LastKnownAddresses but the index must be one-based incrementing positive integers, hence 1, 2, etc.
+    val lastKnownAddressIndex1 = lastKnownAddresses[lastKnownAddresses.indexOfFirst { it.index == 1 }]
+    val lastKnownAddressIndex2 = lastKnownAddresses[lastKnownAddresses.indexOfFirst { it.index == 2 }]
+    assertThat(lastKnownAddressIndex1.lastKnownAddressId.toString(), equalTo(lastKnownAddressId1.toString()))
+    assertThat(lastKnownAddressIndex2.lastKnownAddressId.toString(), equalTo(lastKnownAddressId2.toString()))
```

Comment:
Odd that you need to toString() the UUID

Reply:
Fixed